onnxruntime/core/framework/session_state_utils.cc

```diff
@@ -81,6 +81,11 @@ static common::Status ExtDataTensorProtoToTensor(const Env& env,
   ORT_RETURN_IF_ERROR(utils::GetExtDataFromTensorProto(env, proto_path.c_str(), tensor_proto,
                                                        ext_data_buf, ext_data_len, ext_data_deleter,
                                                        buffered_tensor, &prepacked_for_graph));
+  if constexpr (endian::native != endian::little) {
+    if (!proto_path.empty() && proto_path.compare(onnxruntime::utils::kTensorProtoMemoryAddressTag)) {
```

Comment:
>e(onnxruntime::utils::kTensorProtoMemoryAddressTag) [](http://example.com/codeflow?start=48&length=51)

This tag may mean that TensorProto points to an OrtValue in memory, not necessarily coming form a file.
In that case, the data is already in the proper format.
See buffered tensors.

Reply:
There is work in progress that would convert all initializers into OrtValues, so all the TensorProtos would have that tag.

Reply:
The call stack is like the following:
```
FinalizeSessionState
  -> FinalizeSessionStateImpl
    -> SaveInitializedTensors
      -> DeserializeTensorProto
        -> ExtDataTensorProtoToTensor
```
So it is called after graph optimization during session initialization, and before saving optimized model.

@yuslepukhin, what is your proposed change?

Reply:
I a second thought, the change looks fine, but for readability I suggest comapre() != 0